### PR TITLE
Move mobile test runners to libs.pretest instead of P2P

### PR DESCRIPTION
**Pull request by safern**
Follow up feedback from: https://github.com/dotnet/runtime/pull/36411

### CODE REVIEW

File: src/libraries/Common/tests/AndroidTestRunner/AndroidTestRunner.cs [hunk 1]
```diff
@@ -92,7 +92,9 @@ protected override IEnumerable<TestAssemblyInfo> GetTestAssemblies()
 
     public string? Locale { get; }
 
+#pragma warning disable CS8764
```
Comment (safern) [inline src/libraries/Common/tests/AndroidTestRunner/AndroidTestRunner.cs:95]: This is due to: https://github.com/dotnet/xharness/pull/198 -- once there is a xharness build with that change we can remove this.

**Review by ViktorHofer — APPROVED** [review]
Thanks :)

**Review by steveisok — APPROVED** [review]

File: src/libraries/Common/tests/AndroidTestRunner/AndroidTestRunner.cs (hunk 1 above)
Comment (safern) [reply src/libraries/Common/tests/AndroidTestRunner/AndroidTestRunner.cs:95]: I'll fix this in a separate PR to update xharness.

File: eng/testing/tests.mobile.targets [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <!-- OutDir is not set early enough to set this property in .props file. -->
     <BundleDir>$([MSBuild]::NormalizeDirectory('$(OutDir)', 'AppBundle'))</BundleDir>
     <RunScriptOutputPath>$([MSBuild]::NormalizePath('$(BundleDir)', '$(RunScriptOutputName)'))</RunScriptOutputPath>
-    <RunAOTCompilation Condition="'$(TargetOS)' == 'iOS' and ('$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm')">true</RunAOTCompilation>
+    <RunAOTCompilation Condition="'$(TargetOS)' == 'iOS' and $(TargetArchitecture.StartsWith('arm'))">true</RunAOTCompilation>
```
Comment (akoeplinger) [inline eng/testing/tests.mobile.targets:6]: nit: I prefer using explicit `arm64` and `arm` as it's easier to grep for it

Comment (safern) [reply eng/testing/tests.mobile.targets:6]: I can put it back as it was, I just feel like it is simpler to read the condition. Either way I’m fine.

**Pull request merged**